internal/platform/mmap_linux.go

```diff
@@ -0,0 +1,43 @@
+package platform
+
+import (
+	"syscall"
+	"unsafe"
+)
+
+const (
+	__MREMAP_MAYMOVE = 1
+)
+
+func remapCodeSegmentAMD64(code []byte, size int) ([]byte, error) {
+	return remapCodeSegment(code, size)
+}
+
+func remapCodeSegmentARM64(code []byte, size int) ([]byte, error) {
+	return remapCodeSegment(code, size)
+}
+
+func remapCodeSegment(code []byte, size int) ([]byte, error) {
+	p, err := mremap(*(*unsafe.Pointer)(unsafe.Pointer(&code)), len(code), size, __MREMAP_MAYMOVE)
+	if err != nil {
+		return nil, err
+	}
+	return unsafe.Slice((*byte)(p), size), nil
+}
+
+//go:nosplit
+func mremap(oldAddr unsafe.Pointer, oldSize, newSize, flags int) (unsafe.Pointer, error) {
```

Comment:
is linux the only one you can use remap in of our supported platforms for compiler (freebsd, darwin, windows)?

Reply:
I was only familiar with Linux and assumed BSDs would all look alike but it seems FreeBSD has it https://man.freebsd.org/cgi/man.cgi?query=mremap&manpath=NetBSD+5.0 and obviously the signature is different than Linux 😅